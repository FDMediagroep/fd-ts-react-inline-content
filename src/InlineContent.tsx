import {BulletPoints, BulletPointsStyle} from './components/BulletPoints';
import {LinkBlock, LinkBlockStyle} from './components/LinkBlock';
import {NumberFrame, NumberFrameStyle} from './components/NumberFrame';
import {Quote, QuoteStyle} from './components/Quote';
import {StackFrame, StackFrameStyle, StackFrameDataType} from './components/StackFrame';
import {Stock, StockStyle} from './components/Stock';
import {Summary, SummaryStyle} from './components/Summary';
import {TextFrame, TextFrameStyle} from './components/TextFrame';

export {BulletPoints, BulletPointsStyle};
export {LinkBlock, LinkBlockStyle};
export {NumberFrame, NumberFrameStyle};
export {Quote, QuoteStyle};
export {StackFrame, StackFrameStyle, StackFrameDataType};
export {Stock, StockStyle};
export {Summary, SummaryStyle};
export {TextFrame, TextFrameStyle};
