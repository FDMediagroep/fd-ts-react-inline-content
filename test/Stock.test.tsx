import React from 'react';
import TestRenderer from 'react-test-renderer';
import { Stock } from '../src/InlineContent';

describe('Stock', ()  => {
    it('should render positive correctly', () => {
        let component = TestRenderer.create(<Stock name="INGA" price={10.92} difference={1.90} exchange="XAMS" isin="NL0011821202" url="https://beurs.fd.nl/noteringen/11773/ing/koersen"/>);
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('should render negative correctly', () => {
        let component = TestRenderer.create(<Stock name="INGA" price={10.92} difference={-1.90} exchange="XAMS" isin="NL0011821202" url="https://beurs.fd.nl/noteringen/11773/ing/koersen"/>);
        expect(component.toJSON()).toMatchSnapshot();
    });

});
