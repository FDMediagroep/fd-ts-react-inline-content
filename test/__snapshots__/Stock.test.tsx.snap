// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stock should render negative correctly 1`] = `
<a
  className="company-quote"
  data-exchange="XAMS"
  data-isin="NL0011821202"
  href="https://beurs.fd.nl/noteringen/11773/ing/koersen"
  target="_blank"
>
  <span>
    INGA
  </span>
  <span>
    €
    10.92
  </span>
  <span
    className="drop"
  >
    
    -1.90
    %
  </span>
</a>
`;

exports[`Stock should render positive correctly 1`] = `
<a
  className="company-quote"
  data-exchange="XAMS"
  data-isin="NL0011821202"
  href="https://beurs.fd.nl/noteringen/11773/ing/koersen"
  target="_blank"
>
  <span>
    INGA
  </span>
  <span>
    €
    10.92
  </span>
  <span
    className="climb"
  >
    +
    1.90
    %
  </span>
</a>
`;
